spec.bs

```diff
@@ -427,10 +427,325 @@ The <dfn>standard pathname options</dfn> is an [=options=] [=struct=] with [=opt
+          1. Else if the result of running [=is a hash prefix=] given |parser| is true, then run [=change state=] given |parser|, "<a for="constructor string parser/state">`hash`</a>", and 1.
+        </dd>
+        <dt>"<a for="constructor string parser/state">`port`</a>"</dt>
+        <dd>
+          1. If the result of running [=is a pathname start=] given |parser| is true, then run [=change state=] given |parser|, "<a for="constructor string parser/state">`pathname`</a>", and 0.
+          1. Else if the result of running [=is a search prefix=] given |parser| is true, then run [=change state=] given |parser|, "<a for="constructor string parser/state">`search`</a>", and 1.
+          1. Else if the result of running [=is a hash prefix=] given |parser| is true, then run [=change state=] given |parser|, "<a for="constructor string parser/state">`hash`</a>", and 1.
+        </dd>
+        <dt>"<a for="constructor string parser/state">`pathname`</a>"</dt>
+        <dd>
+          1. If the result of running [=is a search prefix=] given |parser| is true, then run [=change state=] given |parser|, "<a for="constructor string parser/state">`search`</a>", and 1.
+          1. Else if the result of running [=is a hash prefix=] given |parser| is true, then run [=change state=] given |parser|, "<a for="constructor string parser/state">`hash`</a>", and 1.
+        </dd>
+        <dt>"<a for="constructor string parser/state">`search`</a>"</dt>
+        <dd>
+          1. If the result of running [=is a hash prefix=] given |parser| is true, then run [=change state=] given |parser|, "<a for="constructor string parser/state">`hash`</a>", and 1.
+        </dd>
+        <dt>"<a for="constructor string parser/state">`hash`</a>"</dt>
+        <dd>
+          1. Do nothing.
+        </dd>
+        <dt>"<a for="constructor string parser/state">`done`</a>"</dt>
+        <dd>
+          1. [=Assert=]: This step is never reached.
+        </dd>
+      </dl>
+    1. Increment |parser|'s [=constructor string parser/token index=] by |parser|'s [=constructor string parser/token increment=].
+  1. Return |parser|'s [=constructor string parser/result=].
+</div>
+
+<div algorithm>
+To <dfn>change state</dfn> given a [=constructor string parser=] |parser|, a [=constructor string parser/state=] |state|, and a number |skip|:
+
+  1. If |state| is not "<a for="constructor string parser/state">`init`</a>", not "<a for="constructor string parser/state">`authority`</a>", and not "<a for="constructor string parser/state">`done`</a>", then set |parser|'s [=constructor string parser/result=][|state|] to the result of running [=make a component string=] given |parser|.
+  1. Run [=change state without setting the component=] given |parser|, |state|, and |skip|.
+</div>
+
+<div algorithm>
+To <dfn>change state without setting the component</dfn> give a [=constructor string parser=] |parser|, a [=constructor string parser/state=] |state|, and a number |skip|:
```

Comment:
This only has the previous algorithm as a caller, so probably is best inlined into it?

Reply:
Huh, its not called separately in the c++ either.  It definitely was before.  I must have gotten rid of that callsite in a refactor at some point.  I'll inline here.